File: frontend/types/store.ts
export type Store = {
  id: number;
  name: string;
  nuvemshop_store_id: string;
  created_at: string;
};

export type StoreCreateInput = {
  name: string;
  nuvemshop_store_id: string;
  access_token: string;
};

export type Category = {
  id: number;
  name: string;
};

File: backend/app/models/store.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Store(Base):
    __tablename__ = "stores"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    nuvemshop_store_id: Mapped[str] = mapped_column(String(50), nullable=False, unique=True)
    access_token: Mapped[str] = mapped_column(String(255), nullable=False)
    collections_root_category_id: Mapped[int | None] = mapped_column(Integer, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
